src/transformers/models/idefics/modeling_idefics.py

```diff
@@ -65,28 +66,36 @@ def expand_inputs_for_generation(
     )
     input_ids = input_ids.index_select(0, expanded_return_idx)
 
+    model_kwargs["image_encoder_embeddings"] = model_kwargs.get("image_encoder_embeddings", None)
+    model_kwargs["perceiver_embeddings"] = model_kwargs.get("perceiver_embeddings", None)
+    model_kwargs["image_attention_mask"] = model_kwargs.get("image_attention_mask", None)
+
     if "token_type_ids" in model_kwargs:
         token_type_ids = model_kwargs["token_type_ids"]
         model_kwargs["token_type_ids"] = token_type_ids.index_select(0, expanded_return_idx)
 
     if attention_mask is not None:
         model_kwargs["attention_mask"] = attention_mask.index_select(0, expanded_return_idx)
+
+    if model_kwargs["image_attention_mask"] is not None:
         model_kwargs["image_attention_mask"] = model_kwargs["image_attention_mask"].index_select(
             0, expanded_return_idx
         )
-        model_kwargs["pixel_values"] = model_kwargs["pixel_values"].index_select(0, expanded_return_idx)
 
-    if is_encoder_decoder:
-        if encoder_outputs is None:
-            raise ValueError("If `is_encoder_decoder` is True, make sure that `encoder_outputs` is defined.")
-        encoder_outputs["last_hidden_state"] = encoder_outputs.last_hidden_state.index_select(
-            0, expanded_return_idx.to(encoder_outputs.last_hidden_state.device)
+    if model_kwargs["image_encoder_embeddings"] is not None:
+        model_kwargs["image_encoder_embeddings"] = model_kwargs["image_encoder_embeddings"].index_select(
+            0, expanded_return_idx
+        )
+
+    elif model_kwargs["perceiver_embeddings"] is not None:
+        model_kwargs["perceiver_embeddings"] = model_kwargs["perceiver_embeddings"].index_select(
+            0, expanded_return_idx
```

Comment:
Is this something that only happens with `generate`? (if yes, then no worries let's keep it here!)

Reply:
Not sure what you mean here.. Basically, to the best of my understanding, this is used in beam_search as you need to expand the inputs bsz to match the number of beams

Reply:
Ok 👍🏻 then this is all good 😉

Reply:
Actually you if you put outputs in `encoder_outputs` the default `https://github.com/ArthurZucker/transformers/blob/main/src/transformers/generation/utils.py#L721` should work!